src/Compilers/CSharp/Portable/Binder/SwitchBinder.cs

```diff
@@ -267,73 +253,62 @@ private BoundExpression ConvertCaseExpression(TypeSymbol switchGoverningType, CS
                     diagnostics.Add(ErrorCode.WRN_GotoCaseShouldConvert, node.Location, switchGoverningType);
```

Comment:
If we can distinguish between C# 6 and C# 7 switch statements, can we be spec-compliant in the latter?